File: Easy/2894. Divisible and Non-divisible Sums Difference/Python3/main.py
import math
class Solution:
    def differenceOfSums(self, n: int, m: int) -> int:
        total_sum = (n * (n + 1)) / 2
        count = math.floor(n/m)
        divisible_sum = m * (count * (count + 1)) / 2
        return math.floor(total_sum - 2 * divisible_sum)

File: Easy/2894. Divisible and Non-divisible Sums Difference/JavaScript/main.js
/**
 * @param {number} n
 * @param {number} m
 * @return {number}
 */
var differenceOfSums = function (n, m) {
  // Sum of all numbers from 1 to n
  const totalSum = (n * (n + 1)) / 2;

  // Number of multiples of m from 1 to n
  const count = Math.floor(n / m);

  // Sum of all numbers divisible by m from 1 to n
  const divisibleSum = m * (count * (count + 1)) / 2;

  // Return the result
  return totalSum - 2 * divisibleSum;
};


console.log(differenceOfSums(10, 3))
console.log(differenceOfSums(5, 6))
